gateleen-queue/src/main/java/org/swisspush/gateleen/queue/queuing/splitter/QueueSplitterConfigurationParser.java

```diff
@@ -60,9 +61,10 @@ static List<QueueSplitterConfiguration> parse(Buffer configurationResourceBuffer
                             null
                     ));
                 } else {
-                    String postfixFromHeader = queueConfig.getString(POSTFIX_FROM_HEADER_KEY);
-                    String postfixFromUrl = queueConfig.getString(POSTFIX_FROM_URL_KEY);
-                    if (postfixFromHeader != null || postfixFromUrl != null) {
+                    JsonObject postfixFromRequest = queueConfig.getJsonObject(POSTFIX_FROM_REQUEST_KEY);
+                    String postfixFromHeader = postfixFromRequest != null ? postfixFromRequest.getString(POSTFIX_FROM_HEADER_KEY) : null;
+                    String postfixFromUrl = postfixFromRequest != null ? postfixFromRequest.getString(POSTFIX_FROM_URL_KEY) : null;
+                    if (postfixFromRequest != null && (postfixFromHeader != null || postfixFromUrl != null)) {
```

Comment:
`postfixFromRequest != null` can be omitted since you already check `postfixFromHeader` and `postfixFromUrl`